### PR TITLE
cleanup: enable clang-tidy for `storage/tests`

### PR DESCRIPTION
**Pull request by mr-salty**
part of #3958

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/3982)

### CODE REVIEW

**Comment by mr-salty** [comment]
___
*[google/cloud/storage/tests/signed_url_conformance_test.cc, line 327 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/3982#-M6DEXIS5aiN7BhmMSxp:-M6DEXIT8bl83aZNta-Y:b-l2ugxx) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1d5b93a141f7a6f42cf6b0f1ed4a8e6c6e2dbea5/google/cloud/storage/tests/signed_url_conformance_test.cc#L327)):*
> ```
> #if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
>     try
> #endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
> ```

Note to reviewers: is this the right approach here?  clang-tidy warned that `main()` could throw an exception, and `noex` won't build without the `#ifdef`s.  Is `NOLINT` better, or something I am not considering?

File: google/cloud/storage/tests/signed_url_conformance_test.cc [hunk 1]
```diff
@@ -459,4 +459,7 @@ int main(int argc, char* argv[]) {
   ::testing::InitGoogleMock(&argc, argv);
 
   return RUN_ALL_TESTS();
+} catch (std::exception const& ex) {
+  std::cerr << "Standard C++ exception raised: " << ex.what() << "\n";
+  return 1;
```
Comment (devbww) [inline google/cloud/storage/tests/signed_url_conformance_test.cc]: Q. What warning led to this?

Comment (devbww) [reply google/cloud/storage/tests/signed_url_conformance_test.cc]: Just saw your followup from while I was reading.  I'll reply there is needed.

File: google/cloud/storage/tests/curl_request_integration_test.cc [hunk 2]
```diff
@@ -366,9 +366,10 @@ TEST(CurlRequestTest, WellKnownQueryParametersMultiple) {
...
 class MockLogBackend : public google::cloud::LogBackend {
  public:
-  void Process(google::cloud::LogRecord const& lr) override {
-    ProcessWithOwnership(lr);
-  }
+  // TODO(#3983) clang-tidy and clang-3.8 conflict about annotating
+  // `Process` as `override`. See the issue for details.
+  // NOLINTNEXTLINE(modernize-use-override)
+  void Process(google::cloud::LogRecord const& lr) { ProcessWithOwnership(lr); }
   MOCK_METHOD1(ProcessWithOwnership, void(google::cloud::LogRecord));
```
Comment (devjgm) [inline google/cloud/storage/tests/curl_request_integration_test.cc:373]: I think you may be able to fix both warnings and avoid the NOLINT if we change `MOCK_METHOD1` a  `MOCK_METHOD` call:

```cc
  MOCK_METHOD(void, ProcessWithOwnership, (google::cloud::LogRecord), (override));
```

**Pull request merged**